src/test/java/httpserver/router/RouterTest.java

```diff
@@ -41,4 +30,38 @@ void returnsRoutesForGivenRequest() {
 
         assertEquals("[GET, HEAD]", methods);
     }
+
+    @Test
+    void returnsTrueIfPathExistsInRouter() {
+        Request request = new Request(Methods.GET,"/simple_get", Constants.PROTOCOL, null, null);
+        Router router = new Router();
+
+        assertTrue(router.isPathValid(request));
+    }
+
+    @Test
+    void returnsFalseIfPathDoesNotExistInRouter() {
+        Request request = new Request(Methods.GET,"/simple_get_thousand", Constants.PROTOCOL, null, null);
+        Router router = new Router();
+
+        assertFalse(router.isPathValid(request));
+    }
+
+    @Test
+    void returnsTrueIfMethodIsAllowedForRequest() {
+        Request request = new Request(Methods.GET,"/simple_get", Constants.PROTOCOL, null, null);
+        Router router = new Router();
+        String[] methods = router.getMethods(request);
+
+        assertTrue(router.isMethodValid(request, methods));
+    }
+
+    @Test
+    void returnsFalseIfMethodIsNotAllowedForRequest() {
```

Comment:
I try to include both positive and negative tests to ensure the logic is sound, as you did here. Nice work.

Reply:
Learned from the best. Thanks, Rani!